// rxjs-shell/dist/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.listenTerminating = exports.ShellError = exports.spawnEnd = exports.execWithStdin = exports.throwIfStderr = exports.throwIfStdout = exports.throwIf = exports.trim = void 0;
__exportStar(require("./exec"), exports);
__exportStar(require("./execFile"), exports);
__exportStar(require("./fork"), exports);
__exportStar(require("./spawn"), exports);
__exportStar(require("./models"), exports);
var operators_1 = require("./operators");
Object.defineProperty(exports, "trim", { enumerable: true, get: function () { return operators_1.trim; } });
Object.defineProperty(exports, "throwIf", { enumerable: true, get: function () { return operators_1.throwIf; } });
Object.defineProperty(exports, "throwIfStdout", { enumerable: true, get: function () { return operators_1.throwIfStdout; } });
Object.defineProperty(exports, "throwIfStderr", { enumerable: true, get: function () { return operators_1.throwIfStderr; } });
Object.defineProperty(exports, "execWithStdin", { enumerable: true, get: function () { return operators_1.execWithStdin; } });
var util_1 = require("./util");
Object.defineProperty(exports, "spawnEnd", { enumerable: true, get: function () { return util_1.spawnEnd; } });
Object.defineProperty(exports, "ShellError", { enumerable: true, get: function () { return util_1.ShellError; } });
Object.defineProperty(exports, "listenTerminating", { enumerable: true, get: function () { return util_1.listenTerminating; } });
